### PR TITLE
[macos][nativewindowing] Add support for mediakeys

### PR DESCRIPTION
**Pull request by enen92**
## Description
This PR adds support for media keys (keyboard play/pause, rewind, forward) for macOS native windowing.

## Motivation and context
It has come as a feature request in https://github.com/xbmc/xbmc/pull/22990. Not sure if currently the SDL version supports media keys (I guess it doesn't - there's code for it but my testing did not succeed).

## How has this been tested?
Runtime tested. Played a file, hit the different keys.
It might open apple music by default, a few ideas to disable this behaviour globally are given in https://apple.stackexchange.com/questions/380126/do-not-open-apple-music-when-pressing-a-media-key
Unfortunately not something we can do much about.

## What is the effect on users?
Kodi should now respond to media keys.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/windowing/osx/WinEventsOSXImpl.mm [hunk 1]
```diff
@@ -389,4 +409,52 @@ - (BOOL)SendXBMCMouseButtonEvent:(NSEvent*)nsEvent
...
+- (void)SendPlayerAction:(const int&)actionId
+{
+  CAction* action = new CAction(actionId);
+  CServiceBroker::GetAppMessenger()->PostMsg(TMSG_GUI_ACTION, WINDOW_INVALID, -1,
+                                             static_cast<void*>(action));
+}
+
+- (BOOL)HandleMediaKey:(int)keyCode
+{
+  BOOL passEvent = true;
```
Comment (kambala-decapitator) [inline xbmc/windowing/osx/WinEventsOSXImpl.mm:421]:
```suggestion
- (bool)HandleMediaKey:(int)keyCode
{
  bool passEvent = true;
```

also, maybe set it to false and set to true only in `default`?

File: xbmc/windowing/osx/WinEventsOSXImpl.mm [hunk 2]
```diff
@@ -389,4 +409,52 @@ - (BOOL)SendXBMCMouseButtonEvent:(NSEvent*)nsEvent
   return location;
 }
 
+- (void)SendPlayerAction:(const int&)actionId
```
Comment (kambala-decapitator) [inline xbmc/windowing/osx/WinEventsOSXImpl.mm:412]:
```suggestion
- (void)SendPlayerAction:(int)actionId
```

**Review by kambala-decapitator — APPROVED** [review]
will try to make runtime test today

File: xbmc/windowing/osx/WinEventsOSXImpl.mm (hunk 1 above)
Comment (enen92) [reply xbmc/windowing/osx/WinEventsOSXImpl.mm:421]: May I ask why bool instead of BOOL here?

Comment (kambala-decapitator) [reply xbmc/windowing/osx/WinEventsOSXImpl.mm]: the calling code expects C++ bool, not sure why you want to use objc BOOL

**Comment by kambala-decapitator** [comment]
works great!

**Pull request merged**